Automattic/jetpack
# Add Mapkit support in the editor

Fixes https://github.com/Automattic/wp-calypso/issues/73646

## Proposed changes:
This PR adds Apple Maps' Mapkit support in the editor. It does this by checking the `getMapProvider` and conditionally loading either Mapkit or Mapbox. Depending on the map provider, a different component is rendered (`MapkitComponent` or `MapboxComponent`).

These components consume the same properties, but the `MapkitComponent` is written in more modern React, with hooks & context.

<img width="776" alt="CleanShot 2023-03-02 at 14 32 29@2x" src="https://user-images.githubusercontent.com/528287/222443112-2af01040-d5f9-4d90-94b4-31ea73c7ff66.png">

There are still some things outstanding, but this PR is becoming too big already, so I will add those in different PR's:

[] In `extensions/blocks/map/edit.js` a Mapbox API key is still being fetched
[] `extensions/blocks/map/edit.js` has support for a deprecated attribute named `address`. When this is present it geocodes this (using Mapbox), and adds it as a point.
[] There are still unnecessary rerenders. 
[] When editing a point, it doesn't recenter the map on the points if you don't change any field
[] Refactor `mapkit-utils` to `mapkit/utils`
[] Changing map height does not reflect on the frontend

### Other information:


## Jetpack product discussion

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

You should follow these instructions for Mapkit and Mapbox. To force a mapping provider, you can set a cookie by pasting this in your console & refreshing:
- `document.cookie = "map_provider=mapkit; path=/;";`

1. Apply this PR
2. Add or edit a post/page
3. Add the map block
4. Drag the map around, change zoom level
5. Save, the result should be the same on reload
6. Add 1 marker 
7. The map should center around this marker
8. You should still be able to drag & zoom the map, saving it & refreshing gives this result
9. Edit the marker by clicking it, change the title, it should update
10. Add another marker
11. Dragging around & zoom should be disabled now, since it locks to the region
12. Delete one marker

When switching map style to terrain, the mapping provider will switch to Mapbox, no matter what's set in the cookie or query argument. Please verify that this works
